src/fetchcode/package.py

```diff
@@ -362,6 +372,123 @@ def get_gnu_data_from_purl(purl):
     )
 
 
+@router.route("pkg:cocoapods/.*")
+def get_cocoapods_data_from_purl(purl):
+    """
+    Generate `Package` object from the `purl` string of cocoapods type
+    """
+    logging.basicConfig(
+        filename=LOG_FILE_LOCATION,
+        level=logging.WARN,
+        format="%(levelname)s - %(message)s",
+        filemode="w",
+    )
+
+    purl = PackageURL.from_string(purl)
+    name = purl.name
+    version = purl.version
+    cocoapods_org_url = f"https://cocoapods.org/pods/{name}"
+    repository_homepage_url = f"https://cocoapods.org/pods/{name}"
+
+    purl_to_cocoapods_org_url_status = get_cocoapods_org_url_status(purl, name, cocoapods_org_url)
+    cocoa_org_url_status = purl_to_cocoapods_org_url_status["return_message"]
+
+    status_values = [
```

Comment:
Do we have all these statuses in other package implementations?
I am not sure we need to track all these. Having errors and exceptions bubble up is fine here, especially in a low level library. If we need to do more refined handling of errors when we fetch, this should not be something special to cocoapods IMHO.

Reply:
These have been removed along with all related logging, messaging etc.